servant/src/Servant/Links.hs

```diff
@@ -187,6 +189,10 @@ import           Web.HttpApiData
 -- | A safe link datatype.
 -- The only way of constructing a 'Link' is using 'safeLink', which means any
 -- 'Link' is guaranteed to be part of the mentioned API.
+--
+-- NOTE: If you are writing a custom 'HasLink' instance, and need to manipulate
+-- the 'Link' (adding query params or fragments, perhaps), please use the the
+-- 'addQueryParam' and 'addSegment' functions.
```

Comment:
Good Note, thanks!

Reply:
@tchoutri should we also export `addSegment`? Which would force one to also export `Fragment'`? Is there any reason why one would be extending / modifying this behaviour?

Reply:
Not in this PR, I would like to take the time to think about a different module to export these "unsafe" modifiers for Links.